bench-tps/src/cli.rs

```diff
@@ -35,12 +35,14 @@ impl Default for ExternalClientType {
     }
 }
 
+#[derive(PartialEq, Debug)]
```

Comment:
*minor*

When deriving [`PartialEq`](https://doc.rust-lang.org/beta/std/cmp/trait.PartialEq.html), you normally also want to derive [`Eq`](https://doc.rust-lang.org/beta/std/cmp/trait.Eq.html).
Only a rather specific set of types (such as floating point numbers) implement only `PartialEq` without `Eq`.
But as the compiler can not check it, it is up to the person writing the code to make this call.

Here are the details on when absence of `Eq` matters: https://doc.rust-lang.org/book/appendix-03-derivable-traits.html?highlight=partialeq#partialeq-and-eq-for-equality-comparisons

Reply:
Here the problem with `Eq` is that `Keypair` is not implementing it

Reply:
Not sure why it is the case.
`Keypair` is just a `PublicKey` and a `SecretKey`.
Seems like it might be intentional, as `PublicKey` [does implement it](https://docs.rs/ed25519-dalek/latest/ed25519_dalek/struct.PublicKey.html#impl-Eq).
Not sure why [`SecretKey`](https://docs.rs/ed25519-dalek/latest/ed25519_dalek/struct.SecretKey.html) would not.
I've [opened an issue, with a few questions](https://github.com/dalek-cryptography/ed25519-dalek/issues/280), in the library repo.

But you can still add `Eq` to the other places where you are adding `PartialEq`.
At least, I personally, try to do it by default.
When someone needs it, they need to go and figure out if `Eq` was omitted by accident or not.  Which is generally more complex, compared to adding 3 characters when the code is written :))

Reply:
I just realized that `PartialEq` was implemented for `Keypair` in https://github.com/solana-labs/solana/blob/master/sdk/src/signer/keypair.rs#L107-L114
Seems like this is exactly the case when `PartialEq` was implemented, without adding `Eq` in there.

Here is a PR to fix it: https://github.com/solana-labs/solana/pull/30381

Reply:
Looks like leaving as it is